parlai/core/agents.py

```diff
@@ -375,21 +375,39 @@ def load_agent_module(opt):
     model_file = opt['model_file']
     optfile = model_file + '.opt'
     if os.path.isfile(optfile):
+        # Note: new_opt is a bit of a misnomer: opt is what comes from the CLI args,
+        # and new_opt is the options on disk
         new_opt = _load_opt_file(optfile)
         if 'batchindex' in new_opt:
             # This saved variable can cause trouble if we switch to BS=1 at test time
             del new_opt['batchindex']
+
+        # we need to potentially upgrade any old options we might have
+        assert 'model' in new_opt, "Old .opt file doesn't have a model!?"
+        if opt.get('model'):
+            # if the model was specified in the CLI options, use that
+            model_class = get_agent_module(opt['model'])
+            if hasattr(model_class, 'upgrade_saved_opt'):
```

Comment:
you can move these two lines outside of the if/else statement since they are repeated in each case